File: ruby/resistor-color-duo/resistor_color_duo.rb
=begin
Write your code for the 'Resistor Color Duo' exercise in this file. Make the tests in
`resistor_color_duo_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/resistor-color-duo` directory.
=end

class ResistorColorDuo
  def self.value(colors)
    result = colors.map { |color| NUMBERS[color] }
    (result[0].to_s + result[1].to_s).to_i
  end

  NUMBERS = {
    "black"   => 0,
    "brown"   => 1,
    "red"     => 2,
    "orange"  => 3,
    "yellow"  => 4,
    "green"   => 5,
    "blue"    => 6,
    "violet"  => 7,
    "grey"    => 8,
    "white"   => 9
  }
end

File: javascript/resistor-color-duo/resistor-color-duo.js
//
// This is only a SKELETON file for the 'Resistor Color Duo' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

import { symlinkSync } from "fs";

export const decodedValue = (array) => {
  let result = array.map(function(color) {
    if (color == 'black') {
      return 0;
    } else if (color == 'brown') {
      return 1;
    } else if (color == 'red') {
      return 2;
    } else if (color == 'orange') {
      return 3;
    } else if (color == 'yellow') {
      return 4;
    } else if (color == 'green') {
      return 5;
    } else if (color == 'blue') {
      return 6;
    } else if (color == 'violet') {
      return 7;
    } else if (color == 'grey') {
      return 8;
    } else if (color == 'white') {
      return 9;
    }
  });
  return Number(result[0].toString() + result[1].toString())
  // throw new Error("Remove this statement and implement this function");`
};
